tests/stress/stress_test.go

```diff
@@ -1,39 +1,298 @@
+		serviceIDs := strings.Split(terraform.Output("service_ids"), ",\n")
+		methodsToBenchmark["GetService"] = func() { layer0.GetService(serviceIDs[0]) }
+		methodsToBenchmark["ListServices"] = func() { layer0.ListServices() }
+	}
+
+	if c.NumTasks > 0 {
+		methodsToBenchmark["ListTasks"] = func() { layer0.ListTasks() }
+
+		deployIDs := strings.Split(terraform.Output("deploy_ids"), ",\n")
+		environmentIDs := strings.Split(terraform.Output("environment_ids"), ",\n")
+
+		for i := 0; i < c.NumTasks; i++ {
+			environmentID := environmentIDs[i%len(environmentIDs)]
+			deployID := deployIDs[i%len(deployIDs)]
+			taskName := fmt.Sprintf("tt%v", i)
+			layer0.CreateTask(taskName, environmentID, deployID, 1, nil)
+		}
+	}
+
+	benchmark(b, methodsToBenchmark)
+}
+
+func benchmark(b *testing.B, methods map[string]func()) {
+	for name, fn := range methods {
+		b.Run(name, func(b *testing.B) {
+			for n := 0; n < b.N; n++ {
+				fn()
+			}
+		})
+	}
+}
+
+func BenchmarkMinFamiliesMinDeploys(b *testing.B) {
+	runTest(b, StressTestCase{
+		NumDeployFamilies: DEPLOY_FAMILY_SCALE_MIN,
+		NumDeploys:        DEPLOY_SCALE_MIN,
+	})
+}
+
+func BenchmarkMinFamiliesMedDeploys(b *testing.B) {
+	runTest(b, StressTestCase{
+		NumDeployFamilies: DEPLOY_FAMILY_SCALE_MIN,
+		NumDeploys:        DEPLOY_SCALE_MED,
+	})
+}
+
+func BenchmarkMinFamiliesMaxDeploys(b *testing.B) {
+	runTest(b, StressTestCase{
+		NumDeployFamilies: DEPLOY_FAMILY_SCALE_MIN,
+		NumDeploys:        DEPLOY_SCALE_MAX,
+	})
+}
+
+func BenchmarkMedFamiliesMinDeploys(b *testing.B) {
+	runTest(b, StressTestCase{
+		NumDeployFamilies: DEPLOY_FAMILY_SCALE_MED,
+		NumDeploys:        DEPLOY_SCALE_MIN,
+	})
+}
+
+func BenchmarkMedFamiliesMedDeploys(b *testing.B) {
+	runTest(b, StressTestCase{
+		NumDeployFamilies: DEPLOY_FAMILY_SCALE_MED,
+		NumDeploys:        DEPLOY_SCALE_MED,
+	})
+}
+
+func BenchmarkMedFamiliesMaxDeploys(b *testing.B) {
+	runTest(b, StressTestCase{
+		NumDeployFamilies: DEPLOY_FAMILY_SCALE_MED,
+		NumDeploys:        DEPLOY_SCALE_MAX,
+	})
+}
+
+func BenchmarkMaxFamiliesMinDeploys(b *testing.B) {
+	runTest(b, StressTestCase{
+		NumDeployFamilies: DEPLOY_FAMILY_SCALE_MAX,
+		NumDeploys:        DEPLOY_SCALE_MIN,
+	})
+}
+
+func BenchmarkMaxFamiliesMedDeploys(b *testing.B) {
+	runTest(b, StressTestCase{
+		NumDeployFamilies: DEPLOY_FAMILY_SCALE_MAX,
+		NumDeploys:        DEPLOY_SCALE_MED,
+	})
+}
+
+func BenchmarkMaxFamiliesMaxDeploys(b *testing.B) {
+	runTest(b, StressTestCase{
+		NumDeployFamilies: DEPLOY_FAMILY_SCALE_MAX,
+		NumDeploys:        DEPLOY_SCALE_MAX,
+	})
+}
+
+func BenchmarkMinEnvironments(b *testing.B) {
+	runTest(b, StressTestCase{
+		NumEnvironments: ENVIRONMENT_SCALE_MIN,
+	})
+}
+
+func BenchmarkMedEnvironments(b *testing.B) {
+	runTest(b, StressTestCase{
+		NumEnvironments: ENVIRONMENT_SCALE_MED,
+	})
+}
+
+func BenchmarkMaxEnvironments(b *testing.B) {
+	runTest(b, StressTestCase{
+		NumEnvironments: ENVIRONMENT_SCALE_MAX,
+	})
+}
+
+func BenchmarkMinServices(b *testing.B) {
```

Comment:
don't these tests need 1 environment and 1 deploy?

Reply:
They do, but I've built fixes for missing dependencies into the start of `runTest()`. I don't know if we like the pattern, but it trades explicitness in the Benchmark test construction for a tighter focus on what the test is intended to be testing.

Reply:
:+1: